studio/src/pages/RequestorPage/CollectionSection/CollectionItemListItem.tsx

```diff
@@ -0,0 +1,160 @@
+            collectionId: collectionId.toString(),
+            itemId: itemId.toString(),
+          }),
+          search: searchParams.toString(),
+        }}
+        className="grid gap-2 px-2 grid-cols-[4rem_1fr_1fr] rounded-md items-center hover:bg-muted"
+      >
+        <Method className="text-sm" method={route.method} />
+        <span className="text-sm font-mono">{route.path}</span>
+        <span className="text-sm">{name}</span>
+      </Link>
+      <Dialog open={confirmDelete} onOpenChange={setConfirmDelete}>
+        <ConfirmationDialog
+          title={
+            <>
+              Delete{" "}
+              <span className="text-muted-foreground">{name || "no name"}</span>{" "}
+              from collection?
+            </>
+          }
+          description="This action cannot be undone."
+          confirmText="Delete"
+          onConfirm={() => {
+            deleteItem({
+              itemId,
+            });
+            setConfirmDelete(false);
+          }}
+          onCancel={() => setConfirmDelete(false)}
+        />
+      </Dialog>
+      <Dialog open={renameOpen} onOpenChange={setRenameOpen}>
+        <DialogContent className="max-w-screen-sm w-96">
+          <NamingRouteForm
+            error={renameError}
+            isPending={isRenamePending}
+            initialData={{
+              name,
+            }}
+            onSubmit={(data) => {
+              renameItem({
+                itemId,
+                collectionId,
+                extraParams: data,
+              });
+              setRenameOpen(false);
+            }}
+          />
+        </DialogContent>
+      </Dialog>
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button variant={"secondary"} size="icon-xs">
+            <Icon icon="lucide:ellipsis" className="h-3 w-3" />
+          </Button>
+        </DropdownMenuTrigger>
+        <DropdownMenuContent align="end">
+          <CustomDropdownItem onSelect={() => setRenameOpen(true)}>
+            <span>Rename</span>
+            <Icon icon="lucide:pencil" className="h-4 w-4" />
+          </CustomDropdownItem>
+          <CustomDropdownItem onSelect={() => setConfirmDelete(true)}>
+            <span>Delete</span>
+            <Icon icon="lucide:trash-2" className="h-4 w-4" />
+          </CustomDropdownItem>
+          <CustomDropdownItem
+            onSelect={() => {
+              updateItem({
+                collectionId,
+                itemId,
+                extraParams: {
+                  position: 0,
+                },
+              });
+            }}
+          >
+            <span>Move to top</span>
+            <Icon icon="lucide:arrow-up-to-line" className="h-4 w-4" />
+          </CustomDropdownItem>
+          <CustomDropdownItem
+            onSelect={() => {
+              updateItem({
+                collectionId,
+                itemId,
+                extraParams: {
+                  position: Number.MAX_SAFE_INTEGER,
```

Comment:
this causes problems 😅  - it will always be at the bottom, even when i try to reorder something below it

Reply:
okay, issue was on the api side! fixed